SEP/SEP-0009/sep-0009.md

```diff
@@ -0,0 +1,39 @@
+# SPARQL Extensions for Composite Datatypes (Lists and Maps)
+
+* Short name: SPARQL CDTs
+* SEP Number: [SEP-0009](sep-0009.md)
+* Author: [Olaf Hartig](https://github.com/hartig) (joint work with [Gregory Todd Williams](https://github.com/kasei) and other members of the Amazon Neptune team)
+
+Related repository: https://github.com/awslabs/SPARQL-CDTs
+
+## Abstract
+This proposal introduces an approach to represent generic forms of composite values (lists and maps, in particular) as literals in RDF, and corresponding extensions of the SPARQL language. These extensions include an aggregation function to produce such composite values, functions to operate on such composite values in expressions, and a new operator to transform such composite values into their individual components. 
+
+## Motivation
+In contrast to many other popular data representation forms and their query languages, RDF and SPARQL lack built-in support for generic types of composite values such as lists and maps. Instead, RDF introduces so-called [containers](https://www.w3.org/TR/rdf-schema/#ch_containervocab) and [collections](https://www.w3.org/TR/rdf-schema/#ch_collectionvocab), which allow users to model composite values through a dedicated vocabulary _on top_ of the core data model. This approach has several drawbacks when compared to the alternative of representing composite values as compact, self-contained objects. Namely, it can easily become verbose and bloat up the storage footprint; moreover, extracting information from such containers and collections in SPARQL queries is cumbersome and can even be tricky (e.g., enumerating the elements of an RDF collection in their given order requires a complex query if the size of the collection is not assumed to be known), and manipulating containers and collections in SPARQL is complex (e.g., inserting a value into an RDF collection at a particular position is all but trivial).
```

Comment:
```suggestion
In contrast to many other popular data representation forms and their query languages, RDF and SPARQL lack built-in support for generic types of composite values such as lists and maps. Instead, RDF introduces so-called [containers](https://www.w3.org/TR/rdf-schema/#ch_containervocab) and [collections](https://www.w3.org/TR/rdf-schema/#ch_collectionvocab), which allow users to model composite values through a dedicated vocabulary _on top_ of the core data model. This approach has several drawbacks when compared to the alternative of representing composite values as compact, self-contained objects. Especially, it can easily become verbose and bloat the storage footprint; moreover, extracting information from such containers and collections in SPARQL queries is cumbersome and can even be tricky (e.g., enumerating the elements of an RDF collection in their given order requires a complex query, if the size of the collection is not assumed to be known), and manipulating containers and collections in SPARQL is complex (e.g., inserting a value into an RDF collection at a particular position is all but trivial).
```

Reply:
I'm not seeing a change when looking in the GH UI.

Reply:
It's because there are no line breaks, so it's harder for git to tell what changed - the semantic line breaks website gives a good argument why to split lines in docs like this one IMHO https://sembr.org/

Reply:
@afs -- yeah, sorry, GitHub fails to highlight changes on longer "lines" (among other highlight failures). You should be able to use your preferred DIFF tool with a couple copy+pastes.